tests/azure_iot_hub_client_ut.c

```diff
@@ -381,6 +474,690 @@ static void testAzureIoTHubClient_SendTelemetry_Success( void ** state )
+                                                                &requestId ),
+                      eAzureIoTHubClientSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_InvalidArgFailure( void ** state )
+{
+    ( void ) state;
+
+    /* Fail GetDeviceTwin when client is NULL */
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( NULL ),
+                      eAzureIoTHubClientInvalidArgument );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_NotSubscribeFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( &xTestIoTHubClient ),
+                      eAzureIoTHubClientTopicNotSubscribed );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_SendFailure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Subscribe, eAzureIoTMQTTSuccess );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = azureiotmqttPACKET_TYPE_SUBACK;
+    xDeserializedInfo.usPacketIdentifier = usTestPacketId;
+    assert_int_equal( AzureIoTHubClient_SubscribeDeviceTwin( &xTestIoTHubClient,
+                                                             prvTestDeviceTwin,
+                                                             NULL, ( uint32_t ) -1 ),
+                      eAzureIoTHubClientSuccess );
+
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSendFailed );
+    pucPublishPayload = NULL;
+    assert_int_equal( AzureIoTHubClient_GetDeviceTwin( &xTestIoTHubClient ),
+                      eAzureIoTHubClientPublishFailed );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_DeviceTwinGet_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    prvSetupTestIoTHubClient( &xTestIoTHubClient );
+
+    will_return( AzureIoTMQTT_Subscribe, eAzureIoTMQTTSuccess );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = azureiotmqttPACKET_TYPE_SUBACK;
+    xDeserializedInfo.usPacketIdentifier = usTestPacketId;
+    assert_int_equal( AzureIoTHubClient_SubscribeDeviceTwin( &xTestIoTHubClient,
+                                                             prvTestDeviceTwin,
+                                                             NULL, ( uint32_t ) -1 ),
+                      eAzureIoTHubClientSuccess );
+
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSuccess );
+    pucPublishPayload = NULL;
```

Comment:
what are these used for? It's not immediately obvious

Reply:
Test MQTT port has check ... if the payload generated via middleware is equal to the test expected payload. Thich check happens only if test has set the expected MQTT payload.

Reply:
https://github.com/Azure/azure-iot-middleware-freertos/blob/33db1d22451dec6b07302de55c9ba70c711f0f30/tests/azure_iot_cmocka_mqtt.c#L94